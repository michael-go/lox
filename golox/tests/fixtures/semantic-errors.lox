fun readVarInInitializer() {
  var x = x;
}

///

fun duplicateLocalDecleration() {
  var y = "first";
  var y = "second";
}

///

return "bye";
