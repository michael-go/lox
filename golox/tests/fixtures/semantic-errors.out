# exit code: 1
# stdout:

# stderr:
[line 2] Error: Can't read local variable in its own initializer.
[line 9] Error: Already a variable with this name in this scope.
[line 14] Error: Can't return from top-level code.
exit status 65

